libs/wingsdk/src/fs/fs.ts

```diff
@@ -0,0 +1,302 @@
+import * as fs from "fs";
+import * as os from "os";
+import * as path from "path";
+import * as yaml from "js-yaml";
+import { InflightClient } from "../core";
+import { Json } from "../std";
+
+/**
+ * Convert the path to Windows format if running on Windows.
+ * @param p The path to convert.
+ * @returns The Normalized path
+ */
+function localePath(p: string): string {
+  if (process.platform !== "win32") {
+    return p;
+  }
+  let winPath = p.split(path.posix.sep).join(path.win32.sep);
+  // If the path starts from the root directory, the first segment represents a specific drive.
+  winPath = winPath.replace(/^\\([A-Za-z])\\/, "$1:\\");
+  return winPath;
+}
+
+/**
+ * Convert the path to POSIX format.
+ * @param p The path to convert.
+ * @returns The Normalized path.
+ */
+function posixPath(p: string): string {
+  // If the path begins with a specific drive, convert the path to a format that starts from the root directory.
+  // And the initial segment specifies the name of the specific drive.
+  p = p.replace(/^([A-Za-z]):/, "\\$1");
+  return p.split(path.sep).join(path.posix.sep);
+}
+
+/**
+ * Custom settings for reading file
+ */
+export interface ReadFileOptions {
+  /**
+   * The character encoding utilized for file reading.
+   * @default "utf-8"
+   */
+  readonly encoding?: BufferEncoding;
+  /**
+   * The `flag` can be set to specify the attributes.
+   * @default 'r'.
+   */
+  readonly flag?: string;
+}
+
+/**
+ * Custom settings for creating directory
+ */
+export interface MkdirOptions {
+  /**
+   * Indicates whether parent folders should be created.
+   * If a folder was created, the path to the first created folder will be returned.
+   * @default false
+   */
+  readonly recursive?: boolean;
+  /**
+   * A file mode. The string will be parsed as an octal integer.
+   * @default "0o777"
+   */
+  readonly mode?: string;
+}
+
+/**
+ * Custom settings for removing files and directories
+ */
+export interface RemoveOptions {
+  /**
+   * When `true`, exceptions will be ignored if `path` does not exist.
+   * @default false
+   */
+  readonly force?: boolean;
+  /**
+   * If `true`, perform a recursive directory removal. In
+   * recursive mode, operations are retried on failure.
+   * @default false
+   */
+  readonly recursive?: boolean;
+}
+
+/**
+ * The fs class is used for interacting with the file system.
+ * All file paths must be POSIX file paths (/ instead of \),
+ * and will be normalized to the target platform if running on Windows.
+ */
+export class Util {
+  /**
+   * Join all arguments together and normalize the resulting path.
+   * @param paths The array of path need to join.
+   * @returns The resulting path after joining all the paths.
+   */
+  public static join(...paths: string[]): string {
+    return posixPath(path.join(...paths));
```

Comment:
the SDK already has a function for normalizing path separators you can use, in the util file I believe.

```suggestion
    return normalPath(path.join(...paths));
```

Reply:
Thanks for the reminder. However, I wonder if it will work correctly when the path starts with a specified drive. The `mkdtemp` method will return a path with a specified drive. I tried using this method to convert 'C:\a\b\c\d', and the result was 'C:/a/b/c/d'. Does this behave as expected?

Reply:
On windows `C:\a\b\c\d` and `C:/a/b/c/d` are equivalent as far as I know, so I'd say it works as expected if the goal is to at least attempt to normalize between windows and posix.
The normalization is most useful for relative paths of course, which I think will be the normal use-case within wing.